File: 01-Programming-Languages/Go/Data-Structures/LinkedList/linked_list.go
package main

import "fmt"

type Node struct {
    Value int
    Next  *Node
}

type LinkedList struct {
    Head *Node
}

func (ll *LinkedList) Append(value int) {
    newNode := &Node{Value: value}
    
    if ll.Head == nil {
        ll.Head = newNode
        return
    }
    
    current := ll.Head
    for current.Next != nil {
        current = current.Next
    }
    current.Next = newNode
}

func (ll *LinkedList) Prepend(value int) {
    newNode := &Node{Value: value, Next: ll.Head}
    ll.Head = newNode
}

func (ll *LinkedList) Delete(value int) {
    if ll.Head == nil {
        return
    }
    
    if ll.Head.Value == value {
        ll.Head = ll.Head.Next
        return
    }
    
    current := ll.Head
    for current.Next != nil && current.Next.Value != value {
        current = current.Next
    }
    
    if current.Next != nil {
        current.Next = current.Next.Next
    }
}

func (ll *LinkedList) Display() {
    current := ll.Head
    for current != nil {
        fmt.Printf("%d -> ", current.Value)
        current = current.Next
    }
    fmt.Println("nil")
}

func main() {
    ll := &LinkedList{}
    ll.Append(1)
    ll.Append(2)
    ll.Append(3)
    ll.Prepend(0)
    
    fmt.Println("List:")
    ll.Display()
    
    ll.Delete(2)
    fmt.Println("After deleting 2:")
    ll.Display()
}

File: 01-Programming-Languages/Java/Data-Structures/LinkedList/linked_list.java
public class LinkedList {
    private Node head;
    
    private static class Node {
        int value;
        Node next;
        
        Node(int value) {
            this.value = value;
            this.next = null;
        }
    }
    
    public void append(int value) {
        Node newNode = new Node(value);
        
        if (head == null) {
            head = newNode;
            return;
        }
        
        Node current = head;
        while (current.next != null) {
            current = current.next;
        }
        current.next = newNode;
    }
    
    public void prepend(int value) {
        Node newNode = new Node(value);
        newNode.next = head;
        head = newNode;
    }
    
    public void delete(int value) {
        if (head == null) return;
        
        if (head.value == value) {
            head = head.next;
            return;
        }
        
        Node current = head;
        while (current.next != null && current.next.value != value) {
            current = current.next;
        }
        
        if (current.next != null) {
            current.next = current.next.next;
        }
    }
    
    public void display() {
        Node current = head;
        while (current != null) {
            System.out.print(current.value + " -> ");
            current = current.next;
        }
        System.out.println("null");
    }
    
    public static void main(String[] args) {
        LinkedList list = new LinkedList();
        list.append(1);
        list.append(2);
        list.append(3);
        list.prepend(0);
        
        System.out.println("List:");
        list.display();
        
        list.delete(2);
        System.out.println("After deleting 2:");
        list.display();
    }
}